File: frontend/lib/models/organization.dart
/// A company, as a record in its own right.
///
/// Replaces the free-text company field on a contact: one row per company, so
/// "everyone at ACME" is a question with an answer and two spellings are no
/// longer two companies.
class Organization {
  const Organization({
    required this.id,
    required this.name,
    this.domain,
    this.email,
    this.phone,
    this.address,
    this.industry,
    this.notes,
    this.contactCount = 0,
  });

  final String id;
  final String name;
  final String? domain;

  /// The shared mailbox — office@, info@ — rather than a person's own address.
  final String? email;

  /// The switchboard number, for the same reason.
  final String? phone;
  final String? address;
  final String? industry;
  final String? notes;

  /// How many contacts point here. Sent by the API so a list row does not need
  /// a request of its own.
  final int contactCount;

  factory Organization.fromJson(Map<String, dynamic> json) => Organization(
        id: json['id'] as String,
        name: json['name'] as String,
        domain: json['domain'] as String?,
        email: json['email'] as String?,
        phone: json['phone'] as String?,
        address: json['address'] as String?,
        industry: json['industry'] as String?,
        notes: json['notes'] as String?,
        contactCount: json['contact_count'] as int? ?? 0,
      );
}

File: backend/app/schemas/organization.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class OrganizationCreate(BaseModel):
    name: str
    domain: str | None = None
    # Company-level contact details: the shared mailbox and the switchboard,
    # e.g. office@acme.example, rather than a person's own address.
    email: str | None = None
    phone: str | None = None
    address: str | None = None
    industry: str | None = None
    notes: str | None = None


# PATCH uses the same fields but all optional — only sent fields are updated
class OrganizationUpdate(BaseModel):
    name: str | None = None
    domain: str | None = None
    email: str | None = None
    phone: str | None = None
    address: str | None = None
    industry: str | None = None
    notes: str | None = None


class OrganizationRead(OrganizationCreate):
    id: UUID
    # How many contacts point here. Saves the list UI a request per row, and
    # answers "is this company worth keeping?" before a delete.
    contact_count: int
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
